build.gradle.kts

```diff
@@ -0,0 +1,210 @@
+
+    // annotation processor
+    compile("org.atteo.classindex", "classindex", "3.4")
+    apt("org.atteo.classindex", "classindex", "3.4")
+
+    // misc
+    compile("com.adobe.xmp", "xmpcore", "5.1.2")
+    compile("com.drewnoakes", "metadata-extractor", "2.9.1")
+    compile("net.objecthunter", "exp4j", "0.4.8")
+    compile("org.atteo", "evo-inflector", "1.2.2")
+    compile("com.thoughtworks.xstream", "xstream", "1.4.10")
+
+    //	compile("com.twelvemonkeys.imageio", "imageio-core", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-bmp", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-jpeg", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-iff", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-icns", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-pcx", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-pict", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-clippath", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-hdr", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-pdf", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-pnm", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-psd", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-tga", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-sgi", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-thumbsdb", "3.3.2")
+    //	compile("com.twelvemonkeys.imageio", "imageio-tiff", "3.3.2")
+
+    compile(files(file("extra/lib").listFiles()))
+
+}
+
+// source directories
+java.sourceSets {
+    getByName("main") {
+        java.srcDir("src")
+        resources.srcDir("src")
+    }
+}
+
+tasks {
+    val main = "_Main"
+
+    withType<JavaCompile> {
+        options.isWarnings = false
+        options.isDeprecation = false
+        // javac args
+        options.compilerArgs = listOf("-Xlint:unchecked", "-nowarn",
+                "--add-exports", "javafx.graphics/com.sun.javafx.tk=ALL-UNNAMED",
+                "--add-exports", "javafx.graphics/com.sun.javafx.scene.traversal=ALL-UNNAMED",
+                "--add-exports", "javafx.web/com.sun.webkit=ALL-UNNAMED",
+                "--add-exports", "javafx.graphics/com.sun.glass.ui=ALL-UNNAMED")
+    }
+
+    "compileJava" {
+        doFirst {
+            println("Java version: ${JavaVersion.current()}")
+        }
+    }
+
+    "compileKotlin" {
+        doFirst {
+            println("Kotlin version: $kotlinVersion")
+        }
+    }
+
+    "build" {
+        group = main
+    }
+
+    "run" {
+        group = main
+    }
+
+    val cleanup by creating {
+        group = main
+        description = "Cleans the working dir - currently only deletes logs"
+        doFirst {
+            println("Cleaning up...")
+            file("working dir/log").listFiles { file -> arrayOf("log", "zip").contains(file.extension) }.forEach { it.delete() }
```

Comment:
Logs should be in gitignore and not require this step.

Reply:
it's not about git, but for local cleanup. Keep in mind I am still working on the tasks, but that won't influence or break anything. I thought it maybe neccessary for a release task, which is definitely to come

Reply:
I see. This step would definitely be good idea for release/deployment task. Although not so much for build I think. I'll make up my mind when I work it out locally.

Reply:
it's not in the build. It's just a task that's there and can be invoked by `./gradlew cleanup`